app/frontend/src/features/auth/jail/Jail.tsx

```diff
@@ -0,0 +1,60 @@
+import { Backdrop, makeStyles } from "@material-ui/core";
+import React, { useCallback, useEffect, useState } from "react";
+import { Redirect } from "react-router-dom";
+import Alert from "../../../components/Alert";
+import CircularProgress from "../../../components/CircularProgress";
+import PageTitle from "../../../components/PageTitle";
+import TextBody from "../../../components/TextBody";
+import { JailInfoRes } from "../../../pb/jail_pb";
+import { getJailInfo } from "../../../service/jail";
+import { useAppDispatch, useTypedSelector } from "../../../store";
+import { updateJailStatus } from "./jailActions";
+import TOSSection from "./TOSSection";
+
+const useStyles = makeStyles((theme) => ({
+  bottomMargin: { marginBottom: theme.spacing(4) },
+}));
+
+export default function Jail() {
+  const classes = useStyles(makeStyles);
+
+  const dispatch = useAppDispatch();
+  const isJailed = useTypedSelector((state) => state.auth.jailed);
+  const authError = useTypedSelector((state) => state.auth.error);
+
+  const [loading, setLoading] = useState(false);
+  const [jailInfo, setJailInfo] = useState(null as null | JailInfoRes.AsObject);
```

Comment:
```suggestion
  const [jailInfo, setJailInfo] = useState<null | JailInfoRes.AsObject>(null);
```

Since that's what the type param is designed for? Also a better pattern to get into (even though it doesn't matter as much in this case) as type casting is kinda lying to the compiler and putting more work on us to get things right

Reply:
That's better, that was just from my typescript inexperience :D